pkg/tsdb/mysql/mysql_test.go

```diff
@@ -139,40 +139,41 @@ func TestMySQL(t *testing.T) {
 				So(err, ShouldBeNil)
 				queryResult := resp.Results["A"]
 				So(queryResult.Error, ShouldBeNil)
-
-				column := queryResult.Tables[0].Rows[0]
-
-				So(*column[0].(*int8), ShouldEqual, 1)
-				So(column[1].(string), ShouldEqual, "abc")
-				So(column[2].(string), ShouldEqual, "def")
-				So(*column[3].(*int32), ShouldEqual, 1)
-				So(*column[4].(*int16), ShouldEqual, 10)
-				So(*column[5].(*int64), ShouldEqual, 100)
-				So(*column[6].(*int32), ShouldEqual, 1420070400)
-				So(column[7].(float64), ShouldEqual, 1.11)
-				So(column[8].(float64), ShouldEqual, 2.22)
-				So(*column[9].(*float32), ShouldEqual, 3.33)
-				So(column[10].(time.Time), ShouldHappenWithin, 10*time.Second, time.Now())
-				So(column[11].(time.Time), ShouldHappenWithin, 10*time.Second, time.Now())
-				So(column[12].(string), ShouldEqual, "11:11:11")
-				So(column[13].(int64), ShouldEqual, 2018)
-				So(*column[14].(*[]byte), ShouldHaveSameTypeAs, []byte{1})
-				So(column[15].(string), ShouldEqual, "tinytext")
-				So(column[16].(string), ShouldEqual, "tinyblob")
-				So(column[17].(string), ShouldEqual, "text")
-				So(column[18].(string), ShouldEqual, "blob")
-				So(column[19].(string), ShouldEqual, "mediumtext")
-				So(column[20].(string), ShouldEqual, "mediumblob")
-				So(column[21].(string), ShouldEqual, "longtext")
-				So(column[22].(string), ShouldEqual, "longblob")
-				So(column[23].(string), ShouldEqual, "val2")
-				So(column[24].(string), ShouldEqual, "a,b")
-				So(column[25].(time.Time).Format("2006-01-02T00:00:00Z"), ShouldEqual, time.Now().UTC().Format("2006-01-02T00:00:00Z"))
-				So(column[26].(float64), ShouldEqual, float64(1.514764861123456*1e12))
-				So(column[27], ShouldEqual, nil)
-				So(column[28], ShouldEqual, nil)
-				So(column[29], ShouldEqual, "")
-				So(column[30], ShouldEqual, nil)
+				frames, _ := queryResult.Dataframes.Decoded()
```

Comment:
The data package has `data.FrameTestCompareOptions()` meant to be used with `go-cmp`.  

This way you can construct the frame you are expecting as a whole thing.

For an example (in a table test in this case) see https://github.com/grafana/grafana/blob/05fa9d6ad9dd2eff5bae1e5b82806b2f27a8f7a3/pkg/tsdb/azuremonitor/azuremonitor-datasource_test.go#L444-L446